### PR TITLE
Fix ``undefined-loop-variable`` and ``useless-suppression``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Follow up to https://github.com/PyCQA/pylint/pull/6925#issuecomment-1153199648.

### CODE REVIEW

File: pylint/checkers/variables.py [hunk 1]
```diff
@@ -1353,8 +1353,7 @@ def visit_name(self, node: nodes.Name | nodes.AssignName | nodes.DelName) -> Non
             return
 
         self._undefined_and_used_before_checker(node, stmt)
-        if self._is_undefined_loop_variable_enabled:
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/variables.py:1356]: Can't we also remove the class attribute? It's always going to be a problem for useless suppression, right ?

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pylint/checkers/variables.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/checkers/variables.py:1356]: Yeah! Good catch.

Comment (DanielNoord) [reply pylint/checkers/variables.py:1356]: Let's see if this also works. I haven't run tests locally but I think this should work.

File: doc/whatsnew/2/2.14/full.rst [hunk 2]
```diff
@@ -14,6 +14,8 @@ Release date: TBA
 
   Closes #2270
 
+* Allow suppressing ``undefined-loop-variable`` without raising ``useless-suppression``.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2/2.14/full.rst:17]:
```suggestion
* Allow suppressing ``undefined-loop-variable`` and ``undefined-variable`` without raising ``useless-suppression``.
```
Let's only apply this once the github actions are green for the previous commit:)

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**